boot/core/src/boot/core.clj

```diff
@@ -1313,3 +1313,19 @@
           (require 'boot.repl)
           (def ~pod-id (boot.repl/launch-nrepl '~opts)))
         #(pod/with-eval-in p (.close ~pod-id))))))
+
+(defn launch-socket-server [{:keys [bind port accept]}]
+  (let [opts {:host (or bind "127.0.0.1")
+              :port (or port 0)
+              :name (gensym "socket-server")
+              :accept (or accept 'clojure.core.server/repl)}]
+    (try
+      (require 'clojure.core.server)
+      (catch java.io.FileNotFoundException e
+        (throw (ex-info "Socket server requires Clojure version 1.8.0 or above"
+                        {:version (clojure-version)}
+                        e))))
+    (let [effective-port (-> ((resolve 'clojure.core.server/start-server) opts)
+                             (.getLocalPort))]
+      (doto (io/file ".socket-port") .deleteOnExit (spit effective-port))
+      (util/info "Socket server started on port %s on host %s.\n" effective-port (:host opts)))))
```

Comment:
I think this function should be moved into a namespace that is available within pods. 

And could there be a more useful return value than nil? Maybe the name (which is needed when stopping servers) or a function that stops the server?